lesson_18.py

```diff
@@ -0,0 +1,111 @@
+# Task 1
+class Email:
+
+    def __init__(self,email):
+        self.email = self.validate(email)
+
+    @staticmethod
+    def validate(item):
+        valid_endings = ['mail.com', 'mail.cc', 'mail-archive.com', 'mail.org']
+
+        split_item = item.split('@')
+        part1,part2 = split_item
+
+        if part1[0] == '.' or part1[-1] == '.' or part1[-1] == '-' or '#' in part1:
+
+            raise ValueError('Invalid format of the email')
+
+        elif part2 not in valid_endings:
+
+            raise ValueError('Invalid format of the email')
+        return item
+
+# Task 2
+class Boss:
+
+    def __init__(self, id_: int, name: str, company: str):
+        self.id = id_
+
+        self.name = name
+
+        self.company = company
+
+        self._workers = [] # я правильно розумію, що у цьому випадку ми можемо працювати
+        # тільки через приватні змінні?
+
+    @property
+    def workers(self):
+        return self._workers
+
+    @workers.setter
+    def workers(self,args):
+        # new_workers = [x for x in args if x is Worker]
+        self._workers = self._workers + list(args)
+
+class Worker:
+
+    def __init__(self, id_: int, name: str, company: str, boss: Boss):
+        self.id = id_
+
+        self.name = name
+
+        self.company = company
+
+        self._boss = None
+
+    @property
+    def boss(self):
+        return self._boss
+
+    @boss.setter
+    def boss(self,new_boss):
+        if isinstance(new_boss,Boss):
+            self._boss = new_boss
+        raise ValueError()
+
+class TypeDecorators:
+    def __init__(self,orig_func):
```

Comment:
Операцію краще було б додати тут і трошки переробити call. Я напишу саджешенами
```suggestion
    def __init__(self, do):
```